src/Kiota.Builder/Writers/TypeScript/TypeScriptConventionService.cs

```diff
@@ -52,41 +81,91 @@ public override string GetAccessModifier(AccessModifier access)
         };
     }
 
+    public static bool IsComposedOfPrimitives(CodeComposedTypeBase composedType)
+    {
+        // Primitive values don't have a discriminator property so it should be handled differently if any of the values is primitive
+        return composedType?.Types.All(x => IsPrimitiveType(GetTypescriptTypeString(x, composedType))) ?? false;
+    }
+
     public override string GetParameterSignature(CodeParameter parameter, CodeElement targetElement, LanguageWriter? writer = null)
     {
         ArgumentNullException.ThrowIfNull(parameter);
-        var paramType = GetTypeString(parameter.Type, targetElement);
-        var defaultValueSuffix = (string.IsNullOrEmpty(parameter.DefaultValue), parameter.Kind) switch
+        var paramType = GetTypescriptTypeString(parameter.Type, targetElement, inlineComposedTypeString: true);
+        var isComposedOfPrimitives = GetOriginalComposedType(parameter.Type) is CodeComposedTypeBase composedType && IsComposedOfPrimitives(composedType);
+        var defaultValueSuffix = (string.IsNullOrEmpty(parameter.DefaultValue), parameter.Kind, isComposedOfPrimitives) switch
         {
-            (false, CodeParameterKind.DeserializationTarget) => $" = {parameter.DefaultValue}",
-            (false, _) => $" = {parameter.DefaultValue} as {paramType}",
-            (true, _) => string.Empty,
+            (false, CodeParameterKind.DeserializationTarget, false) => $" = {parameter.DefaultValue}",
+            (false, _, false) => $" = {parameter.DefaultValue} as {paramType}",
+            _ => string.Empty,
         };
-        var (partialPrefix, partialSuffix) = parameter.Kind switch
+        var (partialPrefix, partialSuffix) = (isComposedOfPrimitives, parameter.Kind) switch
         {
-            CodeParameterKind.DeserializationTarget => ("Partial<", ">"),
+            (false, CodeParameterKind.DeserializationTarget) => ("Partial<", ">"),
             _ => (string.Empty, string.Empty),
         };
         return $"{parameter.Name.ToFirstCharacterLowerCase()}{(parameter.Optional && parameter.Type.IsNullable ? "?" : string.Empty)}: {partialPrefix}{paramType}{partialSuffix}{(parameter.Type.IsNullable ? " | undefined" : string.Empty)}{defaultValueSuffix}";
     }
+
     public override string GetTypeString(CodeTypeBase code, CodeElement targetElement, bool includeCollectionInformation = true, LanguageWriter? writer = null)
+    {
+        return GetTypescriptTypeString(code, targetElement, includeCollectionInformation, writer);
+    }
+
+    public static string GetTypescriptTypeString(CodeTypeBase code, CodeElement targetElement, bool includeCollectionInformation = true, LanguageWriter? writer = null, bool inlineComposedTypeString = false)
```

Comment:
Its a helper function for the `GetTypeString` function which I can't modify since it inherited
``` c#
public override string GetTypeString(CodeTypeBase code, CodeElement targetElement, bool includeCollectionInformation = true, LanguageWriter? writer = null)
{
        return GetTypescriptTypeString(code, targetElement, includeCollectionInformation, writer);
}
```
I

Reply:
`GetTypescriptTypeString` isn't inheriting from another function so it doesn't need to carry forward parameters it is not using. Yes?

Reply:
What will happen if in future someone calls `GetTypeString` with full parameters. I am just trying to be consistent with what already exists

Reply:
The helper `GetTypescriptTypeString` isn't using the parameter. So why would we need to keep it/receive it? Those functions should also therefore not be passing it over. Yes?

Reply:
Confirmed as resolved.